### PR TITLE
feat(orders): ORDERS-5715 add OrderFee interface, add fees field to Order interface

### PR DESCRIPTION
**Pull request by matt-evangelidis**
## What?
This PR adds an `OrderFee` interface and a `fees` field to the `Order` interface in `order.ts`.

## Why?
This adds some interfaces for fees specific to the Orders domain to provide separation between Checkout and Orders, as each domain's definitions of fees are likely to drift further apart over time.

## Testing / Proof
N/A

@bigcommerce/team-checkout @bigcommerce/team-orders

### CODE REVIEW

File: packages/core/src/order/order.ts
Comment (animesh1987) [inline packages/core/src/order/order.ts:37]: Is this supposed to an optional?

**Comment by animesh1987** [comment]
@matt-evangelidis  For commit validation the scope needs to be `order` compared to orders 😄

File: packages/core/src/order/order.ts
Comment (matt-evangelidis) [reply packages/core/src/order/order.ts:37]: I marked it as optional because fees are only included by default when coming from `api/storefront/orders/` if `PROJECT-5577.support_order_level_fee` is on.

I suppose that the field itself should always be in the payload, so it should be fine to have as mandatory.

Comment (animesh1987) [reply packages/core/src/order/order.ts:37]: I think that makes sense to mark it mandatory.

Comment (matt-evangelidis) [reply packages/core/src/order/order.ts]: Marked as mandatory now.

**Review by animesh1987 — APPROVED** [review]
Nicely done, looks good. Can you please check if checkout js build works fine with this change before merging.

**Comment by matt-evangelidis** [comment]
> Nicely done, looks good. Can you please check if checkout js build works fine with this change before merging.

image

**Comment by matt-evangelidis** [comment]
@animesh1987 I don't have write access to this repo. Are you alright to merge?

**Pull request merged**
